File: yurongyang/258.cpp
class Solution {
public:
    int addDigits(int num) {
    int result = 0;
        if(num/10==0)
            return num;
        while(num){  
            result+=num%10;
            num = num/10;
        }
    return addDigits(result);
    }
};

File: sparrowsong/258.go
package leetcode

func addDigits(num int) int {
	if num/10 == 0 {
		return num % 10
	}
	ans := 0
	for num > 0 {
		ans += num % 10
		num /= 10
	}
	return addDigits(ans)
}
